engine/schema/src/main/java/com/cloud/vm/VMInstanceVO.java

```diff
@@ -200,7 +200,7 @@
     @Column(name = "backup_external_id")
     protected String backupExternalId;
 
-    @Column(name = "backup_volumes")
+    @Column(name = "backup_volumes", length = 65535)
```

Comment:
@olivierlemasle can you please extract the length value to a constant?
e.g. MAX_VARCHAR_LENGTH = 65535

Reply:
Sure @GabrielBrascher! Do you suggest creating a new `Enum` with all column lenghts used in the codebase?

Because there's already about 130 `@Column`-annotated fields with an explicit length, all of them using such a "magic number" (including 14 fields which already use `length = 65535`!)

```
./engine/storage/src/main/java/org/apache/cloudstack/storage/db/ObjectInDataStoreVO.java:    @Column(name = "url", length = 2048)
./engine/orchestration/src/main/java/org/apache/cloudstack/engine/datacenter/entity/api/db/EngineDataCenterVO.java:    @Column(name = "engine_state", updatable = true, nullable = false, length = 32)
./engine/orchestration/src/main/java/org/apache/cloudstack/engine/datacenter/entity/api/db/EngineClusterVO.java:    @Column(name = "engine_state", updatable = true, nullable = false, length = 32)
./engine/orchestration/src/main/java/org/apache/cloudstack/engine/datacenter/entity/api/db/EngineHostVO.java:    @Column(name = "engine_state", updatable = true, nullable = false, length = 32)
./engine/orchestration/src/main/java/org/apache/cloudstack/engine/datacenter/entity/api/db/EngineHostPodVO.java:    @Column(name = "engine_state", updatable = true, nullable = false, length = 32)
./engine/schema/src/main/java/org/apache/cloudstack/engine/cloud/entity/api/db/VMEntityVO.java:    @Column(name = "name", updatable = false, nullable = false, length = 255)
./engine/schema/src/main/java/org/apache/cloudstack/engine/cloud/entity/api/db/VMEntityVO.java:    @Column(name = "vnc_password", updatable = true, nullable = false, length = 255)
./engine/schema/src/main/java/org/apache/cloudstack/engine/cloud/entity/api/db/VMEntityVO.java:    @Column(name = "state", updatable = true, nullable = false, length = 32)
./engine/schema/src/main/java/org/apache/cloudstack/engine/cloud/entity/api/db/VMEntityVO.java:    @Column(name = "vm_template_id", updatable = true, nullable = true, length = 17)
./engine/schema/src/main/java/org/apache/cloudstack/engine/cloud/entity/api/db/VMEntityVO.java:    @Column(name = "guest_os_id", nullable = false, length = 17)
./engine/schema/src/main/java/org/apache/cloudstack/engine/cloud/entity/api/db/VMEntityVO.java:    @Column(name = "vm_type", updatable = false, nullable = false, length = 32)
./engine/schema/src/main/java/org/apache/cloudstack/direct/download/DirectDownloadCertificateVO.java:    @Column(name = "certificate", length = 65535)
./engine/schema/src/main/java/org/apache/cloudstack/ha/HAConfigVO.java:    @Column(name = "ha_state", updatable = true, nullable = false, length = 32)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/VpcGatewayDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/DiskOfferingDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/NetworkACLItemDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/AutoScaleVmGroupDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/NetworkACLListDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/SnapshotPolicyDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/VpcDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/UserDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/LBStickinessPolicyDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/LBHealthCheckPolicyDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/Site2SiteVpnConnectionDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/FirewallRuleDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/UserIpAddressDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/AutoScaleVmProfileDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/Site2SiteVpnGatewayDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/RemoteAccessVpnDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/Site2SiteCustomerGatewayDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/resourcedetail/GuestOsDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/org/apache/cloudstack/region/gslb/GlobalLoadBalancerRuleVO.java:    @Column(name = "description", length = 4096)
./engine/schema/src/main/java/org/apache/cloudstack/storage/datastore/db/TemplateDataStoreVO.java:    @Column(name = "url", length = 2048)
./engine/schema/src/main/java/org/apache/cloudstack/storage/datastore/db/TemplateDataStoreVO.java:    @Column(name = "download_url", length = 2048)
./engine/schema/src/main/java/org/apache/cloudstack/storage/datastore/db/StoragePoolVO.java:    @Column(name = "name", updatable = false, nullable = false, length = 255)
./engine/schema/src/main/java/org/apache/cloudstack/storage/datastore/db/StoragePoolVO.java:    @Column(name = "uuid", length = 255)
./engine/schema/src/main/java/org/apache/cloudstack/storage/datastore/db/StoragePoolVO.java:    @Column(name = "pool_type", updatable = false, nullable = false, length = 32)
./engine/schema/src/main/java/org/apache/cloudstack/storage/datastore/db/ImageStoreVO.java:    @Column(name = "url", nullable = false, length = 2048)
./engine/schema/src/main/java/org/apache/cloudstack/storage/datastore/db/VolumeDataStoreVO.java:    @Column(name = "url", length = 2048)
./engine/schema/src/main/java/org/apache/cloudstack/storage/datastore/db/VolumeDataStoreVO.java:    @Column(name = "download_url", length = 2048)
./engine/schema/src/main/java/org/apache/cloudstack/lb/ApplicationLoadBalancerRuleVO.java:    @Column(name = "description", length = 4096)
./engine/schema/src/main/java/org/apache/cloudstack/outofbandmanagement/OutOfBandManagementVO.java:    @Column(name = "power_state", updatable = true, nullable = false, length = 32)
./engine/schema/src/main/java/com/cloud/certificate/CertificateVO.java:    @Column(name = "certificate", length = 65535)
./engine/schema/src/main/java/com/cloud/network/as/AutoScaleVmProfileVO.java:    @Column(name = "template_id", updatable = true, nullable = false, length = 17)
./engine/schema/src/main/java/com/cloud/network/as/AutoScaleVmProfileVO.java:    @Column(name = "other_deploy_params", updatable = true, length = 1024)
./engine/schema/src/main/java/com/cloud/network/dao/NetworkVO.java:    @Column(name = "guru_data", length = 1024)
./engine/schema/src/main/java/com/cloud/network/dao/NetworkDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/com/cloud/network/dao/LoadBalancerVO.java:    @Column(name = "description", length = 4096)
./engine/schema/src/main/java/com/cloud/network/dao/SslCertVO.java:    @Column(name = "certificate", length = 16384)
./engine/schema/src/main/java/com/cloud/network/dao/SslCertVO.java:    @Column(name = "chain", length = 2097152)
./engine/schema/src/main/java/com/cloud/network/dao/SslCertVO.java:    @Column(name = "key", length = 16384)
./engine/schema/src/main/java/com/cloud/network/vpc/NetworkACLItemVO.java:    @Column(name = "reason", length = 2500)
./engine/schema/src/main/java/com/cloud/deployasis/TemplateDeployAsIsDetailVO.java:    @Column(name = "value", length = 65535)
./engine/schema/src/main/java/com/cloud/deployasis/UserVmDeployAsIsDetailVO.java:    @Column(name = "value", length = 65535)
./engine/schema/src/main/java/com/cloud/offerings/NetworkOfferingVO.java:    @Column(name = "tags", length = 4096)
./engine/schema/src/main/java/com/cloud/offerings/NetworkOfferingDetailsVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/com/cloud/storage/VolumeHostVO.java:    @Column(name = "url", length = 2048)
./engine/schema/src/main/java/com/cloud/storage/DiskOfferingVO.java:    @Column(name = "display_text", length = 4096)
./engine/schema/src/main/java/com/cloud/storage/DiskOfferingVO.java:    @Column(name = "tags", length = 4096)
./engine/schema/src/main/java/com/cloud/storage/VMTemplateVO.java:    @Column(name = "url", length = 2048)
./engine/schema/src/main/java/com/cloud/storage/VMTemplateVO.java:    @Column(name = "display_text", length = 4096)
./engine/schema/src/main/java/com/cloud/storage/UploadVO.java:    @Column(name = "url", length = 2048)
./engine/schema/src/main/java/com/cloud/storage/VolumeVO.java:    @Column(name = "chain_info", length = 65535)
./engine/schema/src/main/java/com/cloud/storage/VMTemplateDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/com/cloud/storage/VMTemplateHostVO.java:    @Column(name = "url", length = 2048)
./engine/schema/src/main/java/com/cloud/storage/VolumeDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/com/cloud/event/EventVO.java:    @Column(name = "description", length = 1024)
./engine/schema/src/main/java/com/cloud/event/EventVO.java:    @Column(name = "parameters", length = 1024)
./engine/schema/src/main/java/com/cloud/vm/UserVmVO.java:    @Column(name = "iso_id", nullable = true, length = 17)
./engine/schema/src/main/java/com/cloud/vm/UserVmVO.java:    @Column(name = "user_data", updatable = true, nullable = true, length = 32768)
./engine/schema/src/main/java/com/cloud/vm/NicDetailVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/com/cloud/vm/UserVmDetailVO.java:    @Column(name = "value", length = 5120)
./engine/schema/src/main/java/com/cloud/vm/VMInstanceVO.java:    @Column(name = "name", nullable = false, length = 255)
./engine/schema/src/main/java/com/cloud/vm/VMInstanceVO.java:    @Column(name = "vnc_password", updatable = true, nullable = false, length = 255)
./engine/schema/src/main/java/com/cloud/vm/VMInstanceVO.java:    @Column(name = "state", updatable = true, nullable = false, length = 32)
./engine/schema/src/main/java/com/cloud/vm/VMInstanceVO.java:    @Column(name = "vm_template_id", updatable = true, nullable = true, length = 17)
./engine/schema/src/main/java/com/cloud/vm/VMInstanceVO.java:    @Column(name = "guest_os_id", nullable = false, length = 17)
./engine/schema/src/main/java/com/cloud/vm/VMInstanceVO.java:    @Column(name = "vm_type", updatable = false, nullable = false, length = 32)
./engine/schema/src/main/java/com/cloud/user/UserAccountVO.java:    @Column(name = "external_entity", length = 65535)
./engine/schema/src/main/java/com/cloud/user/SSHKeyPairVO.java:    @Column(name = "public_key", length = 5120)
./engine/schema/src/main/java/com/cloud/user/UserVO.java:    @Column(name = "external_entity", length = 65535)
./engine/schema/src/main/java/com/cloud/dc/VlanDetailsVO.java:    @Column(name = "value", length = 1024)
./engine/schema/src/main/java/com/cloud/alert/AlertVO.java:    @Column(name = "subject", length = 999)
./engine/schema/src/main/java/com/cloud/alert/AlertVO.java:    @Column(name = "content", length = 5000)
./plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/KubernetesClusterDetailsVO.java:    @Column(name = "value", length = 10240)
./plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/KubernetesClusterVO.java:    @Column(name = "description", length = 4096)
./server/src/main/java/com/cloud/api/query/vo/UserVmJoinVO.java:    @Column(name = "name", updatable = false, nullable = false, length = 255)
./server/src/main/java/com/cloud/api/query/vo/UserVmJoinVO.java:    @Column(name = "display_name", updatable = false, nullable = false, length = 255)
./server/src/main/java/com/cloud/api/query/vo/UserVmJoinVO.java:    @Column(name = "vm_type", updatable = false, nullable = false, length = 32)
./server/src/main/java/com/cloud/api/query/vo/UserVmJoinVO.java:    @Column(name = "state", updatable = true, nullable = false, length = 32)
./server/src/main/java/com/cloud/api/query/vo/UserVmJoinVO.java:    @Column(name = "guest_os_id", nullable = false, length = 17)
./server/src/main/java/com/cloud/api/query/vo/UserVmJoinVO.java:    @Column(name = "template_id", updatable = true, nullable = true, length = 17)
./server/src/main/java/com/cloud/api/query/vo/UserVmJoinVO.java:    @Column(name = "template_display_text", length = 4096)
./server/src/main/java/com/cloud/api/query/vo/UserVmJoinVO.java:    @Column(name = "iso_id", updatable = true, nullable = true, length = 17)
./server/src/main/java/com/cloud/api/query/vo/UserVmJoinVO.java:    @Column(name = "iso_display_text", length = 4096)
./server/src/main/java/com/cloud/api/query/vo/UserVmJoinVO.java:    @Column(name = "pool_type", updatable = false, nullable = false, length = 32)
./server/src/main/java/com/cloud/api/query/vo/UserVmJoinVO.java:    @Column(name = "user_data", updatable = true, nullable = true, length = 2048)
./server/src/main/java/com/cloud/api/query/vo/TemplateJoinVO.java:    @Column(name = "display_text", length = 4096)
./server/src/main/java/com/cloud/api/query/vo/ImageStoreJoinVO.java:    @Column(name = "url", length = 2048)
./server/src/main/java/com/cloud/api/query/vo/DomainRouterJoinVO.java:    @Column(name = "name", updatable = false, nullable = false, length = 255)
./server/src/main/java/com/cloud/api/query/vo/DomainRouterJoinVO.java:    @Column(name = "state", updatable = true, nullable = false, length = 32)
./server/src/main/java/com/cloud/api/query/vo/DomainRouterJoinVO.java:    @Column(name = "template_id", updatable = true, nullable = true, length = 17)
./server/src/main/java/com/cloud/api/query/vo/AsyncJobJoinVO.java:    @Column(name = "job_result", length = 65535)
./server/src/main/java/com/cloud/api/query/vo/AsyncJobJoinVO.java:    @Column(name = "instance_type", length = 64)
./server/src/main/java/com/cloud/api/query/vo/AsyncJobJoinVO.java:    @Column(name = "instance_id", length = 64)
./server/src/main/java/com/cloud/api/query/vo/DiskOfferingJoinVO.java:    @Column(name = "tags", length = 4096)
./server/src/main/java/com/cloud/api/query/vo/ServiceOfferingJoinVO.java:    @Column(name = "tags", length = 4096)
./server/src/main/java/com/cloud/api/query/vo/EventJoinVO.java:    @Column(name = "parameters", length = 1024)
./server/src/main/java/com/cloud/api/query/vo/VolumeJoinVO.java:    @Column(name = "template_display_text", length = 4096)
./server/src/main/java/com/cloud/api/query/vo/VolumeJoinVO.java:    @Column(name = "iso_id", updatable = true, nullable = true, length = 17)
./server/src/main/java/com/cloud/api/query/vo/VolumeJoinVO.java:    @Column(name = "iso_display_text", length = 4096)
./server/src/main/java/com/cloud/api/query/vo/VolumeJoinVO.java:    @Column(name = "chain_info", length = 65535)
./server/src/main/java/com/cloud/api/query/vo/NetworkOfferingJoinVO.java:    @Column(name = "tags", length = 4096)
./framework/quota/src/main/java/org/apache/cloudstack/quota/vo/UserVmDetailVO.java:    @Column(name = "value", length = 5120)
./framework/security/src/main/java/org/apache/cloudstack/framework/security/keystore/KeystoreVO.java:    @Column(name = "certificate", length = 65535)
./framework/security/src/main/java/org/apache/cloudstack/framework/security/keystore/KeystoreVO.java:    @Column(name = "key", length = 65535)
./framework/jobs/src/main/java/org/apache/cloudstack/framework/jobs/impl/AsyncJobJoinMapVO.java:    @Column(name = "join_result", length = 1024)
./framework/jobs/src/main/java/org/apache/cloudstack/framework/jobs/impl/AsyncJobJournalVO.java:    @Column(name = "journal_type", updatable = false, nullable = false, length = 32)
./framework/jobs/src/main/java/org/apache/cloudstack/framework/jobs/impl/AsyncJobJournalVO.java:    @Column(name = "journal_text", length = 1024)
./framework/jobs/src/main/java/org/apache/cloudstack/framework/jobs/impl/AsyncJobJournalVO.java:    @Column(name = "journal_obj", length = 1024)
./framework/jobs/src/main/java/org/apache/cloudstack/framework/jobs/impl/AsyncJobVO.java:    @Column(name = "job_type", length = 32)
./framework/jobs/src/main/java/org/apache/cloudstack/framework/jobs/impl/AsyncJobVO.java:    @Column(name = "job_dispatcher", length = 64)
./framework/jobs/src/main/java/org/apache/cloudstack/framework/jobs/impl/AsyncJobVO.java:    @Column(name = "job_cmd_info", length = 65535)
./framework/jobs/src/main/java/org/apache/cloudstack/framework/jobs/impl/AsyncJobVO.java:    @Column(name = "job_result", length = 65535)
./framework/jobs/src/main/java/org/apache/cloudstack/framework/jobs/impl/AsyncJobVO.java:    @Column(name = "instance_type", length = 64)
./framework/jobs/src/main/java/org/apache/cloudstack/framework/jobs/impl/AsyncJobVO.java:    @Column(name = "instance_id", length = 64)
./framework/config/src/main/java/org/apache/cloudstack/framework/config/impl/ConfigurationVO.java:    @Column(name = "value", length = 8191)
./framework/config/src/main/java/org/apache/cloudstack/framework/config/impl/ConfigurationVO.java:    @Column(name = "default_value", length = 8191)
./framework/config/src/main/java/org/apache/cloudstack/framework/config/impl/ConfigurationVO.java:    @Column(name = "description", length = 1024)
```

Reply:
That sounds like a plan @olivierlemasle :+1: 

Let me be clear that I am not **-1** if you keep it as it is right now. But I think that it could be interesting to avoid "magical" numbers for length.

I see that normalizing column length values to constants/Enum is the kind of enhancement that might need to be addressed on a PR of its own; therefore, I would not be **-1** with your proposal as it is right now.